Flickr/Sources/FlickrAppVCs/SearchResultsVC.swift

```diff
@@ -6,17 +6,200 @@
+        guard let url = returnSearchUrl(searchString: searchString) else {
+            return
+        }
+        let task = URLSession.shared.dataTask(with: url, completionHandler: { data, _, error in
+            guard let data = data, error == nil else {
+                return
+            }
+            var result: Photos?
+            do {
+                result = try JSONDecoder().decode(Photos.self, from: data)
+            } catch {
+                return
+            }
+            guard let result = result else {
+                return
+            }
+            photosArray = self.constructIndividualUrls(result)
+            self.photos = photosArray
+            DispatchQueue.main.async {
+                self.collectionView?.reloadData()
+            }
+        })
+        task.resume()
+    }
+
+    func returnCollectionViewFlowLayout() -> UICollectionViewFlowLayout {
+        let collectionViewLayout = UICollectionViewFlowLayout()
+        collectionViewLayout.minimumInteritemSpacing = minimumInteritemSpacing
+        collectionViewLayout.minimumLineSpacing = minimumLineSpacing
+        collectionViewLayout.sectionInset = sectionInset
+        collectionViewLayout.itemSize = CGSize(width: view.frame.width / widthDivisor, height: cellHeight)
+        return collectionViewLayout
+    }
+
+    func setupCollectionView() -> [NSLayoutConstraint] {
+        let collectionViewFlowLayout = returnCollectionViewFlowLayout()
+        let collectionView = UICollectionView(frame: view.frame, collectionViewLayout: collectionViewFlowLayout)
+        collectionView.configureView { collectionView in
+            collectionView.delegate = self
+            collectionView.dataSource = self
+            collectionView.register(UICollectionViewCell.self, forCellWithReuseIdentifier: cellReuseIdentifier)
+        }
+        self.collectionView = collectionView
+        view.addSubview(collectionView)
+
+        return [
+            collectionView.leftAnchor.constraint(equalTo: view.leftAnchor),
+            collectionView.rightAnchor.constraint(equalTo: view.rightAnchor),
+            collectionView.topAnchor.constraint(equalTo: view.topAnchor),
+            collectionView.bottomAnchor.constraint(equalTo: view.bottomAnchor)
+        ]
+    }
+
+    func returnSpinner(cell: UICollectionViewCell) -> [NSLayoutConstraint] {
+        let spinner = UIActivityIndicatorView()
+        spinner.configureView { spinner in
+            spinner.startAnimating()
+        }
+        cell.addSubview(spinner)
+        return [
+            spinner.centerXAnchor.constraint(equalTo: cell.centerXAnchor),
+            spinner.centerYAnchor.constraint(equalTo: cell.centerYAnchor)
+        ]
+    }
+}
+
+extension SearchResultsVC: UICollectionViewDataSource, UICollectionViewDelegate, UICollectionViewDelegateFlowLayout {
+    func collectionView(_: UICollectionView, numberOfItemsInSection _: Int) -> Int {
+        photos.isEmpty ? 20 : photos.count
```

Comment:
Why return 20 if no results? Should you not prevent the VC from coming to this screen if there are no results?